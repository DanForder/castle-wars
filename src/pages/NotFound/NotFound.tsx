import { Link } from "react-router-dom";
import Layout from "../../components/Layout/Layout";

const NotFound = () => (
  <Layout>
    <h1>Page Not Found!</h1>
    <Link to="/">Home</Link>
  </Layout>
);

export default NotFound;
