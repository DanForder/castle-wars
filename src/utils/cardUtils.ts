import { CardInfo } from "../types/CardInfo";

export const getCostImage = (type: CardInfo["costType"]): string => {
  return {
    brick: "🧱",
    weapon: "⚔️",
    crystal: "🔮",
  }[type];
};
