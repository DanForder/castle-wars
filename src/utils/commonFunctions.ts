export const getRandomArrayItem = <Type>(array: Type[]): Type => {
  return array[Math.floor(Math.random() * array.length)];
};

export const getRandomIntegerInRange = (min: number, max: number): number => {
  return Math.floor(Math.random() * (max - min + 1) + min);
};
