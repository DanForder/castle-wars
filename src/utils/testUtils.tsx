import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import PlayerNameProvider from "../context/GameContext";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return (
    <PlayerNameProvider>
      <BrowserRouter>{ui}</BrowserRouter>
    </PlayerNameProvider>
  );
};
