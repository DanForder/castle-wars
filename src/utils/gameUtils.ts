import { clamp } from "./commonFunctions";

export const getBuildingHeight = (hp: number): number => {
  const result = hp / 100;

  if (result > 1) {
    return 1;
  }

  if (result < 0) {
    return 0;
  }

  return result;
};

export const getAddBetweenBounds = (
  minBound: number,
  maxBound: number,
  current: number,
  amount: number
): number => {
  const naturalResult = current + amount;
  const clampedResult = clamp(minBound, maxBound, naturalResult);
  const isNegative = Math.sign(amount) === -1;

  if (naturalResult === clampedResult) return amount;

  return isNegative ? minBound - current : maxBound - current;
};
