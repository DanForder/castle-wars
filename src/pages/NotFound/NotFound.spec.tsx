import { customRender } from "../../utils/testUtils";
import NotFound from "./NotFound";

export default describe("NotFound Page", () => {
  it("Renders", () => {
    const { container } = customRender(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
