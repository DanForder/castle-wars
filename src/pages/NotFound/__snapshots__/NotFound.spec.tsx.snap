// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page Renders 1`] = `
<div>
  <h1
    style="color: rgb(37, 37, 42);"
  >
    Page Not Found!
  </h1>
  <a
    href="/"
    style="color: rgb(50, 50, 206);"
  >
    Home
  </a>
</div>
`;
